### DISCUSSION TITLE
Mapping each item of an array in a JSON response to a new line containing other fields from that structure

### DISCUSSION THREAD

**Discussion by eroth** [TomWright/dasel#287] [Q&A] [answered]
Hi, this may be kind of a weird one (or maybe I'm just phrasing it weirdly). Consider the following JSON response from a GH GraphQL query:

```JSON
{
  "data": {
    "search": {
      "nodes": [
        {
          "number": 6666,
          "title": "Some PR title",
          "author": {
            "login": "sgtpepper"
          },
          "createdAt": "2022-12-21T02:35:59Z",
          "comments": {
            "nodes": [
              {
                "author": {
                  "login": "jamesdean"
                },
                "createdAt": "2022-12-21T02:36:40Z",
                "bodyText": "I love this PR"
              },
              {
                "author": {
                  "login": "billybob"
                },
                "createdAt": "2022-12-22T00:04:39Z",
                "bodyText": "This PR is the best thing since sliced bread"
              }
            ]
          }
        }
      ]
    }
  }
}
```
As can be seen, the `comments` array is nested in the response for PR 6666. The CSV output I'd like to see from this would look like:
```CSV
number,title,author,createdAt,commentAuthor,commentText,commentTime
6666,"Some PR title","sgtpepper","2022-12-21T02:35:59Z","jamesdean","I love this PR","2022-12-21T02:36:40Z"
6666,"Some PR title","sgtpepper","2022-12-21T02:35:59Z","billybob","This PR is the best thing since sliced bread","2022-12-22T00:04:39Z"
```
In other words, for each comment node in the `comments` array for this PR node create a new line with `number`, `title`, `author`,`createdAt` with comment info from each comment node appended. Is that something that dasel supports? I didn't really see an example of this. Thanks!

**Comment by TomWright** [comment, marked as answer]
OK so I've got a version of this working:

```
$ dasel -w csv -f /Users/tom/tmp/dasel/287/data.json '.data.search.nodes.all().comments.nodes.all().mapOf(number,parent(3).number,title,parent(3).title,author,parent(3).author.login,createdAt,parent(3).createdAt,commentAuthor,author.login,commentText,bodyText,commentTime,createdAt).merge()'
author,commentAuthor,commentText,commentTime,createdAt,number,title
sgtpepper,jamesdean,I love this PR,2022-12-21T02:36:40Z,2022-12-21T02:35:59Z,6666.000000,Some PR title
sgtpepper,billybob,This PR is the best thing since sliced bread,2022-12-22T00:04:39Z,2022-12-21T02:35:59Z,6666.000000,Some PR title
```

Note that you'll need to use the latest version `v2.1.0` for access to `parent(3)`. You may need to use the development version if homebrew haven't merged the release PR yet.

`parent()` existed before but accepted no arguments, and there's an issue with using `parent().parent().parent()` due to it thinking the parent was the previous parent call.

I have made a new release `v2.1.0` containing this change.

**Reply by eroth** [reply]
Thanks, you're the best! Ah, that makes sense you'd have to traverse down to the deepest node and then back reference nodes higher up using `mapOf`. I won't have a chance to try this out until tomorrow morning my time, but I wanted to thank you so much for looking into this and cutting a release for it 🙌

**Reply by TomWright** [reply]
No problem 🙂 Let me know if you have any issues with it

**Reply by eroth** [reply]
This works like a charm! Thanks so much!
